R/pkg/inst/worker/worker.R

```diff
@@ -84,67 +84,78 @@ broadcastElap <- elapsedSecs()
 # as number of partitions to create.
 numPartitions <- SparkR:::readInt(inputCon)
 
-isDataFrame <- as.logical(SparkR:::readInt(inputCon))
+# 0 - RDD mode, 1 - dapply mode, 2 - gapply mode
+mode <- SparkR:::readInt(inputCon)
 
-# If isDataFrame, then read column names
-if (isDataFrame) {
+# If DataFrame - mode = 1 and mode = 2, then read column names
+if (mode > 0) {
```

Comment:
```
Thnx @sun-rui !
Maybe it will be better to have smth like:
```
isDataFrame = FALSE
dapply = FALSE
gapply = FALSE
if (mode > 0) {
  isDataFrame = TRUE
  if (mode == 1) {
    dapply = TRUE
  } else {
    gapply = TRUE
  }
}
```
because in your example, we'd also need to set, gapply = FALSE and dapply = FALSE accordingly is mode > 0. Smth like:
 if (mode == 1) {
      dapply = TRUE
      gapply = FALSE
    } else {
      dapply = FALSE
      gapply = TRUE
    }
```